nixos/modules/services/misc/sssd.nix

```diff
@@ -60,22 +88,28 @@ in {
         wants = [ "nss-user-lookup.target" ];
         restartTriggers = [
           config.environment.etc."nscd.conf".source
-          config.environment.etc."sssd/sssd.conf".source
+          settingsFileUnsubstituted
         ];
         script = ''
           export LDB_MODULES_PATH+="''${LDB_MODULES_PATH+:}${pkgs.ldb}/modules/ldb:${pkgs.sssd}/modules/ldb"
           mkdir -p /var/lib/sss/{pubconf,db,mc,pipes,gpo_cache,secrets} /var/lib/sss/pipes/private /var/lib/sss/pubconf/krb5.include.d
-          ${pkgs.sssd}/bin/sssd -D
+          ${pkgs.sssd}/bin/sssd -D -c ${settingsFile}
         '';
         serviceConfig = {
           Type = "forking";
           PIDFile = "/run/sssd.pid";
+          StateDirectory = baseNameOf dataDir;
```

Comment:
How does that work when `dataDir` is any path which does not start with `/var/lib`, e.g. `dataDir = /data/sssd`, it would give a `StateDirectory = "sssd";` and creates a folder in `/var/lib/sssd`, right?

Reply:
`dataDir` is hardcoded and not an option.

Reply:
Ha, I stand corrected then.